tests/tests/integration_tests.rs

```diff
@@ -256,3 +256,26 @@ async fn server_should_be_able_to_close_subscriptions() {
 
 	assert!(matches!(res, Err(Error::SubscriptionClosed(_))));
 }
+
+#[tokio::test]
+async fn ws_close_pending_subscription_when_server_terminated() {
+	let (server_addr, mut handle) = websocket_server_with_subscription().await;
+	let server_url = format!("ws://{}", server_addr);
+
+	let c1 = WsClientBuilder::default().build(&server_url).await.unwrap();
+
+	let mut sub: Subscription<String> =
+		c1.subscribe("subscribe_hello", JsonRpcParams::NoParams, "unsubscribe_hello").await.unwrap();
+
+	assert!(matches!(sub.next().await, Ok(Some(_))));
+
+	handle.stop().await.unwrap();
+	handle.wait_for_stop().await;
+
+	let sub2: Result<Subscription<String>, _> =
+		c1.subscribe("subscribe_hello", JsonRpcParams::NoParams, "unsubscribe_hello").await;
+
+	// no new request should be accepted.
+	assert!(matches!(sub2, Err(_)));
+	assert!(matches!(sub.next().await, Ok(None)));
```

Comment:
I think this line is also testing that the first subscription is closed, too?

Reply:
exactly, I can add a comment for this,

It is to ensure that already established subscriptions are canceled too because they are handled a bit different but because the sink is sharing the `receiver` from the background loop and once it's dropped it causes the `sender/sink` [to fail too here](https://github.com/paritytech/jsonrpsee/blob/master/utils/src/server/rpc_module.rs#L391) I realized now that error message is wrong, I can fix that 

`Ok(None)` means that the stream was terminated in the case.

Reply:
That makes sense! A comment would be handy for someone like me, just so it's obvious that the test is also aiming to check that.